src/DynamoCore/Logging/AnalyticsService.cs

```diff
@@ -78,6 +80,12 @@ internal static bool IsADPOptedIn
 
                 adpAnalyticsUI.SetOptedIn(value);
             }
+            
+        }
+
+        internal static bool IsADPAvailable()
+        {
+            return adpAnalyticsUI.IsProviderAvailable();
```

Comment:
you should check that `if (!Analytics.DisableAnalytics && adpAnalyticsUI != null)`